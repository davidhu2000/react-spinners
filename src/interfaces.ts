import { SerializedStyles } from '@emotion/react';

export interface LengthObject {
  value: number;
  unit: string;
}

interface CommonProps {
  color?: string;
  loading?: boolean;
  css?: string | SerializedStyles;
}

export type LengthType = number | string;

export interface LoaderHeightWidthProps extends CommonProps {
  height?: LengthType;
  width?: LengthType;
}

export interface LoaderSizeProps extends CommonProps {
  size?: LengthType;
}

export interface LoaderSizeMarginProps extends LoaderSizeProps {
  margin?: LengthType;
}

export interface LoaderHeightWidthRadiusProps extends LoaderHeightWidthProps {
  margin?: LengthType;
  radius?: LengthType;
}
