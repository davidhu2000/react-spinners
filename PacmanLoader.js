(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['exports', 'emotion', 'react', 'prop-types', '@emotion/core', 'recompose/onlyUpdateForKeys'], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require('emotion'), require('react'), require('prop-types'), require('@emotion/core'), require('recompose/onlyUpdateForKeys'));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.emotion, global.react, global.propTypes, global.core, global.onlyUpdateForKeys);
    global.PacmanLoader = mod.exports;
  }
})(this, function (exports, _emotion, _react, _propTypes, _core, _onlyUpdateForKeys) {
  'use strict';

  Object.defineProperty(exports, "__esModule", {
    value: true
  });

  var _react2 = _interopRequireDefault(_react);

  var _propTypes2 = _interopRequireDefault(_propTypes);

  var _onlyUpdateForKeys2 = _interopRequireDefault(_onlyUpdateForKeys);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  function _classCallCheck(instance, Constructor) {
    if (!(instance instanceof Constructor)) {
      throw new TypeError("Cannot call a class as a function");
    }
  }

  var _createClass = function () {
    function defineProperties(target, props) {
      for (var i = 0; i < props.length; i++) {
        var descriptor = props[i];
        descriptor.enumerable = descriptor.enumerable || false;
        descriptor.configurable = true;
        if ("value" in descriptor) descriptor.writable = true;
        Object.defineProperty(target, descriptor.key, descriptor);
      }
    }

    return function (Constructor, protoProps, staticProps) {
      if (protoProps) defineProperties(Constructor.prototype, protoProps);
      if (staticProps) defineProperties(Constructor, staticProps);
      return Constructor;
    };
  }();

  function _possibleConstructorReturn(self, call) {
    if (!self) {
      throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
    }

    return call && (typeof call === "object" || typeof call === "function") ? call : self;
  }

  function _inherits(subClass, superClass) {
    if (typeof superClass !== "function" && superClass !== null) {
      throw new TypeError("Super expression must either be null or a function, not " + typeof superClass);
    }

    subClass.prototype = Object.create(superClass && superClass.prototype, {
      constructor: {
        value: subClass,
        enumerable: false,
        writable: true,
        configurable: true
      }
    });
    if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass;
  }

  // This returns an animation
  var pacman = [(0, _core.keyframes)('0%{transform:rotate(0deg)}50%{transform:rotate(-44deg)}'), (0, _core.keyframes)('0%{transform:rotate(0deg)}50%{transform:rotate(44deg)}')];

  var Loader = function (_React$Component) {
    _inherits(Loader, _React$Component);

    function Loader() {
      var _ref;

      var _temp, _this, _ret;

      _classCallCheck(this, Loader);

      for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = Loader.__proto__ || Object.getPrototypeOf(Loader)).call.apply(_ref, [this].concat(args))), _this), _this.ball = function () {
        var _this$props = _this.props,
            size = _this$props.size,
            sizeUnit = _this$props.sizeUnit;


        return (0, _core.keyframes)('75%{opacity:0.7}100%{transform:translate(', '' + -4 * size + sizeUnit, ',', '' + -size / 4 + sizeUnit, ')}');
      }, _this.ballStyle = function (i) {
        var _this$props2 = _this.props,
            color = _this$props2.color,
            margin = _this$props2.margin,
            size = _this$props2.size,
            sizeUnit = _this$props2.sizeUnit;


        return (0, _core.css)('{width:', '' + size / 3 + sizeUnit, ';height:', '' + size / 3 + sizeUnit, ';background-color:', color, ';margin:', margin, ';border-radius:100%;transform:translate(0,', '' + -size / 4 + sizeUnit, ');position:absolute;top:', size, 'px;left:', '' + size * 4 + sizeUnit, ';animation:', _this.ball(), ' 1s ', i * 0.25, 's infinite linear;animation-fill-mode:both;}');
      }, _this.s1 = function () {
        var _this$props3 = _this.props,
            size = _this$props3.size,
            sizeUnit = _this$props3.sizeUnit;


        return '' + size + sizeUnit + ' solid transparent';
      }, _this.s2 = function () {
        var _this$props4 = _this.props,
            size = _this$props4.size,
            sizeUnit = _this$props4.sizeUnit,
            color = _this$props4.color;


        return '' + size + sizeUnit + ' solid ' + color;
      }, _this.pacmanStyle = function (i) {
        var _this$props5 = _this.props,
            size = _this$props5.size,
            sizeUnit = _this$props5.sizeUnit;


        var s1 = _this.s1();
        var s2 = _this.s2();

        return (0, _core.css)('{width:0;height:0;border-right:', s1, ';border-top:', i === 0 ? s1 : s2, ';border-left:', s2, ';border-bottom:', i === 0 ? s2 : s1, ';border-radius:', '' + size + sizeUnit, ';position:absolute;animation:', pacman[i], ' 0.8s infinite ease-in-out;animation-fill-mode:both;}');
      }, _this.wrapper = function () {
        var _this$props6 = _this.props,
            size = _this$props6.size,
            sizeUnit = _this$props6.sizeUnit;


        var wrapper = (0, _core.css)('{position:relative;font-size:0;height:', '' + size + sizeUnit, ';width:', '' + size + sizeUnit, ';}');

        return _this.props.css ? (0, _core.css)(wrapper, ';', _this.props.css) : wrapper;
      }, _this.pac = function () {
        return _this.pacmanStyle(0);
      }, _this.man = function () {
        return _this.pacmanStyle(1);
      }, _temp), _possibleConstructorReturn(_this, _ret);
    }

    _createClass(Loader, [{
      key: 'render',
      value: function render() {
        var loading = this.props.loading;


        return loading ? _react2.default.createElement(
          'div',
          {
            className: (0, _emotion.css)(this.wrapper())
          },
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.pac())
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.man())
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.ballStyle(2))
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.ballStyle(3))
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.ballStyle(4))
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.ballStyle(5))
          })
        ) : null;
      }
    }]);

    return Loader;
  }(_react2.default.Component);

  Loader.propTypes = {
    loading: _propTypes2.default.bool,
    color: _propTypes2.default.string,
    size: _propTypes2.default.number,
    margin: _propTypes2.default.string,
    sizeUnit: _propTypes2.default.string,
    css: _propTypes2.default.string
  };

  Loader.defaultProps = {
    loading: true,
    color: '#000000',
    size: 25,
    margin: '2px',
    sizeUnit: 'px',
    css: ''
  };

  var Component = (0, _onlyUpdateForKeys2.default)(['loading', 'color', 'size', 'margin', 'sizeUnit', 'css'])(Loader);
  Component.defaultProps = Loader.defaultProps;
  exports.default = Component;
});