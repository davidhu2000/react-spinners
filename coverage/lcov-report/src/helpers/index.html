
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/helpers</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/helpers</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>86/86</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.88% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>32/36</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>9/9</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>85/85</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="animation.ts"><a href="animation.ts.html">animation.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

<tr>
	<td class="file high" data-value="colors.ts"><a href="colors.ts.html">colors.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="47" class="abs high">47/47</td>
	<td data-value="95" class="pct high">95%</td>
	<td data-value="20" class="abs high">19/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="46" class="abs high">46/46</td>
	</tr>

<tr>
	<td class="file high" data-value="unitConverter.ts"><a href="unitConverter.ts.html">unitConverter.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="10" class="abs high">8/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Tue Jun 25 2024 21:55:12 GMT-0700 (Pacific Daylight Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    