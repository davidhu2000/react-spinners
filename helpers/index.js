(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['exports'], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports);
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports);
    global.index = mod.exports;
  }
})(this, function (exports) {
  'use strict';

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  var calculateRgba = exports.calculateRgba = function calculateRgba(color, opacity) {
    if (color[0] === '#') {
      color = color.slice(1);
    }

    if (color.length === 3) {
      var res = '';
      color.split('').forEach(function (c) {
        res += c;
        res += c;
      });
      color = res;
    }

    var rgbValues = color.match(/.{2}/g).map(function (hex) {
      return parseInt(hex, 16);
    }).join(', ');
    return 'rgba(' + rgbValues + ', ' + opacity + ')';
  };
});