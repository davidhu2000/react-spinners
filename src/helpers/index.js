export const calculateRgba = (color, opacity) => {
  if (color[0] === '#') {
    color = color.slice(1);
  }

  if (color.length === 3) {
    let res = '';
    color.split('').forEach(c => {
      res += c;
      res += c;
    });
    color = res;
  }

  let rgbValues = color.match(/.{2}/g).map(hex => parseInt(hex, 16)).join(', ');
  return `rgba(${rgbValues}, ${opacity})`;
};

export * from './proptypes';
