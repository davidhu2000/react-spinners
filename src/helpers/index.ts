export * from "./proptypes";
export * from "./colors";
export * from "./unitConverter";
