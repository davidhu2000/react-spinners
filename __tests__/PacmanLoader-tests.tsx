import * as React from "react";
import { mount, ReactWrapper } from "enzyme";
import { matchers } from '@emotion/jest';
expect.extend(matchers);

import PacmanLoader from "../src/PacmanLoader";
import { LoaderSizeMarginProps } from "../src/interfaces";
import { sizeMarginDefaults } from "../src/helpers";

describe("PacmanLoader", () => {
  let loader: ReactWrapper;
  let props: LoaderSizeMarginProps;
  const defaultSize = 25;
  const defaultMargin = 2;
  const defaultColor = "#000000";
  const defaultUnit = "px";

  it("should match snapshot", () => {
    loader = mount(<PacmanLoader />);
    expect(loader).toMatchSnapshot();
  });

  it("should contain default props if no props are passed", () => {
    props = loader.props();
    expect(props).toEqual(sizeMarginDefaults(defaultSize));
  });

  it("should contain styles created using default props", () => {
    expect(loader).toHaveStyleRule("height", `${defaultSize}${defaultUnit}`);
    expect(loader).toHaveStyleRule("width", `${defaultSize}${defaultUnit}`);

    expect(loader.find("span span").at(0)).toHaveStyleRule(
      "border-radius",
      `${defaultSize}${defaultUnit}`
    );
    expect(loader.find("span span").at(1)).toHaveStyleRule(
      "border-radius",
      `${defaultSize}${defaultUnit}`
    );

    for (let i = 2; i < 6; i++) {
      expect(loader.find("span span").at(i)).toHaveStyleRule("background-color", defaultColor);
      expect(loader.find("span span").at(i)).toHaveStyleRule(
        "height",
        `${defaultSize / 3}${defaultUnit}`
      );
      expect(loader.find("span span").at(i)).toHaveStyleRule(
        "width",
        `${defaultSize / 3}${defaultUnit}`
      );
      expect(loader.find("span span").at(i)).toHaveStyleRule(
        "margin",
        `${defaultMargin}${defaultUnit}`
      );
      expect(loader.find("span span").at(i)).toHaveStyleRule("top", `${defaultSize}${defaultUnit}`);
      expect(loader.find("span span").at(i)).toHaveStyleRule(
        "left",
        `${defaultSize * 4}${defaultUnit}`
      );
    }
  });

  it("should render null if loading prop is set as false", () => {
    loader = mount(<PacmanLoader loading={false} />);
    expect(loader.isEmptyRender()).toBe(true);
  });

  it("should render the correct color based on prop", () => {
    const color = "#e2e2e2";
    loader = mount(<PacmanLoader color={color} />);
    for (let i = 2; i < 6; i++) {
      expect(loader.find("span span").at(i)).not.toHaveStyleRule("background-color", defaultColor);
      expect(loader.find("span span").at(i)).toHaveStyleRule("background-color", color);
    }
  });

  describe("size props", () => {
    it("should render the size with px unit when size is a number", () => {
      const size = 18;
      loader = mount(<PacmanLoader size={size} />);

      expect(loader).not.toHaveStyleRule("width", `${defaultSize}${defaultUnit}`);
      expect(loader).toHaveStyleRule("width", `${size}${defaultUnit}`);

      expect(loader.find("span span").at(0)).not.toHaveStyleRule(
        "border-radius",
        `${defaultSize / 3}${defaultUnit}`
      );
      expect(loader.find("span span").at(1)).not.toHaveStyleRule(
        "border-radius",
        `${defaultSize / 3}${defaultUnit}`
      );

      expect(loader.find("span span").at(0)).toHaveStyleRule(
        "border-radius",
        `${size}${defaultUnit}`
      );
      expect(loader.find("span span").at(1)).toHaveStyleRule(
        "border-radius",
        `${size}${defaultUnit}`
      );

      for (let i = 2; i < 6; i++) {
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "height",
          `${defaultSize / 3}${defaultUnit}`
        );
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "width",
          `${defaultSize / 3}${defaultUnit}`
        );
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "top",
          `${defaultSize}${defaultUnit}`
        );
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "left",
          `${defaultSize * 4}${defaultUnit}`
        );

        expect(loader.find("span span").at(i)).toHaveStyleRule("height", `${size / 3}${defaultUnit}`);
        expect(loader.find("span span").at(i)).toHaveStyleRule("width", `${size / 3}${defaultUnit}`);
        expect(loader.find("span span").at(i)).toHaveStyleRule("top", `${size}${defaultUnit}`);
        expect(loader.find("span span").at(i)).toHaveStyleRule("left", `${size * 4}${defaultUnit}`);
      }
    });

    it("should render the size as is when size is a string with valid css unit", () => {
      const length = 18;
      const unit = "px";
      const size = `${length}${unit}`;
      loader = mount(<PacmanLoader size={size} />);

      expect(loader).not.toHaveStyleRule("width", `${defaultSize}${defaultUnit}`);
      expect(loader).toHaveStyleRule("width", `${size}`);

      expect(loader.find("span span").at(0)).not.toHaveStyleRule(
        "border-radius",
        `${defaultSize / 3}${defaultUnit}`
      );
      expect(loader.find("span span").at(1)).not.toHaveStyleRule(
        "border-radius",
        `${defaultSize / 3}${defaultUnit}`
      );

      expect(loader.find("span span").at(0)).toHaveStyleRule("border-radius", `${size}`);
      expect(loader.find("span span").at(1)).toHaveStyleRule("border-radius", `${size}`);

      for (let i = 2; i < 6; i++) {
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "height",
          `${defaultSize / 3}${defaultUnit}`
        );
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "width",
          `${defaultSize / 3}${defaultUnit}`
        );
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "top",
          `${defaultSize}${defaultUnit}`
        );
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "left",
          `${defaultSize * 4}${defaultUnit}`
        );

        expect(loader.find("span span").at(i)).toHaveStyleRule("height", `${length / 3}${unit}`);
        expect(loader.find("span span").at(i)).toHaveStyleRule("width", `${length / 3}${unit}`);
        expect(loader.find("span span").at(i)).toHaveStyleRule("top", `${length}${unit}`);
        expect(loader.find("span span").at(i)).toHaveStyleRule("left", `${length * 4}${unit}`);
      }
    });

    it("should render the size with default unit of px when the unit is incorrect", () => {
      const length = 18;
      const unit = "ad";
      const size = `${length}${unit}`;
      loader = mount(<PacmanLoader size={size} />);

      expect(loader).not.toHaveStyleRule("width", `${defaultSize}${defaultUnit}`);
      expect(loader).toHaveStyleRule("width", `${length}${defaultUnit}`);

      expect(loader.find("span span").at(0)).not.toHaveStyleRule(
        "border-radius",
        `${defaultSize / 3}${defaultUnit}`
      );
      expect(loader.find("span span").at(1)).not.toHaveStyleRule(
        "border-radius",
        `${defaultSize / 3}${defaultUnit}`
      );

      expect(loader.find("span span").at(0)).toHaveStyleRule(
        "border-radius",
        `${length}${defaultUnit}`
      );
      expect(loader.find("span span").at(1)).toHaveStyleRule(
        "border-radius",
        `${length}${defaultUnit}`
      );

      for (let i = 2; i < 6; i++) {
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "height",
          `${defaultSize / 3}${defaultUnit}`
        );
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "width",
          `${defaultSize / 3}${defaultUnit}`
        );
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "top",
          `${defaultSize}${defaultUnit}`
        );
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "left",
          `${defaultSize * 4}${defaultUnit}`
        );

        expect(loader.find("span span").at(i)).toHaveStyleRule(
          "height",
          `${length / 3}${defaultUnit}`
        );
        expect(loader.find("span span").at(i)).toHaveStyleRule(
          "width",
          `${length / 3}${defaultUnit}`
        );
        expect(loader.find("span span").at(i)).toHaveStyleRule("top", `${length}${defaultUnit}`);
        expect(loader.find("span span").at(i)).toHaveStyleRule("left", `${length * 4}${defaultUnit}`);
      }
    });
  });

  describe("margin props", () => {
    it("should render the margin with px unit when margin is a number", () => {
      const margin = 18;
      loader = mount(<PacmanLoader margin={margin} />);
      for (let i = 2; i < 6; i++) {
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "margin",
          `${defaultMargin}${defaultUnit}`
        );
        expect(loader.find("span span").at(i)).toHaveStyleRule("margin", `${margin}${defaultUnit}`);
      }
    });

    it("should render the margin as is when margin is a string with valid css unit", () => {
      const length = 18;
      const unit = "px";
      const margin = `${length}${unit}`;

      loader = mount(<PacmanLoader margin={margin} />);
      for (let i = 2; i < 6; i++) {
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "margin",
          `${defaultMargin}${defaultUnit}`
        );
        expect(loader.find("span span").at(i)).toHaveStyleRule("margin", `${margin}`);
      }
    });

    it("should render the margin with default unit of px when the unit is incorrect", () => {
      const length = 18;
      const unit = "ad";
      const margin = `${length}${unit}`;

      loader = mount(<PacmanLoader margin={margin} />);
      for (let i = 2; i < 6; i++) {
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "margin",
          `${defaultMargin}${defaultUnit}`
        );
        expect(loader.find("span span").at(i)).toHaveStyleRule("margin", `${length}${defaultUnit}`);
      }
    });
  });

  it("should render the css override based on props", () => {
    loader = mount(
      <PacmanLoader css={"position: fixed; width: 100px; height: 200px; color: blue;"} />
    );
    expect(loader).not.toHaveStyleRule("position", "relative");
    expect(loader).not.toHaveStyleRule("width", "25px");
    expect(loader).toHaveStyleRule("position", "fixed");
    expect(loader).toHaveStyleRule("width", "100px");
  });
});
