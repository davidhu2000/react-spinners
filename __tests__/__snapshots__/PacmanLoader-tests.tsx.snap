// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PacmanLoader should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  50% {
    -webkit-transform: rotate(-44deg);
    -ms-transform: rotate(-44deg);
    transform: rotate(-44deg);
  }
}

@keyframes animation-1 {
  0% {
    -webkit-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  50% {
    -webkit-transform: rotate(44deg);
    -ms-transform: rotate(44deg);
    transform: rotate(44deg);
  }
}

@keyframes animation-2 {
  75% {
    opacity: 0.7;
  }

  100% {
    -webkit-transform: translate(-100px,-6.25px);
    -ms-transform: translate(-100px,-6.25px);
    transform: translate(-100px,-6.25px);
  }
}

@keyframes animation-2 {
  75% {
    opacity: 0.7;
  }

  100% {
    -webkit-transform: translate(-100px,-6.25px);
    -ms-transform: translate(-100px,-6.25px);
    transform: translate(-100px,-6.25px);
  }
}

@keyframes animation-2 {
  75% {
    opacity: 0.7;
  }

  100% {
    -webkit-transform: translate(-100px,-6.25px);
    -ms-transform: translate(-100px,-6.25px);
    transform: translate(-100px,-6.25px);
  }
}

@keyframes animation-2 {
  75% {
    opacity: 0.7;
  }

  100% {
    -webkit-transform: translate(-100px,-6.25px);
    -ms-transform: translate(-100px,-6.25px);
    transform: translate(-100px,-6.25px);
  }
}

.emotion-6 {
  position: relative;
  font-size: 0;
  height: 25px;
  width: 25px;
}

.emotion-0 {
  width: 0;
  height: 0;
  border-right: 25px solid transparent;
  border-top: 25px solid transparent;
  border-left: 25px solid #000000;
  border-bottom: 25px solid #000000;
  border-radius: 25px;
  position: absolute;
  -webkit-animation: animation-0 0.8s infinite ease-in-out;
  animation: animation-0 0.8s infinite ease-in-out;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.emotion-1 {
  width: 0;
  height: 0;
  border-right: 25px solid transparent;
  border-top: 25px solid #000000;
  border-left: 25px solid #000000;
  border-bottom: 25px solid transparent;
  border-radius: 25px;
  position: absolute;
  -webkit-animation: animation-1 0.8s infinite ease-in-out;
  animation: animation-1 0.8s infinite ease-in-out;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.emotion-2 {
  width: 8.333333333333334px;
  height: 8.333333333333334px;
  background-color: #000000;
  margin: 2px;
  border-radius: 100%;
  -webkit-transform: translate(0,-6.25px);
  -ms-transform: translate(0,-6.25px);
  transform: translate(0,-6.25px);
  position: absolute;
  top: 25px;
  left: 100px;
  -webkit-animation: animation-2 1s 0.5s infinite linear;
  animation: animation-2 1s 0.5s infinite linear;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.emotion-3 {
  width: 8.333333333333334px;
  height: 8.333333333333334px;
  background-color: #000000;
  margin: 2px;
  border-radius: 100%;
  -webkit-transform: translate(0,-6.25px);
  -ms-transform: translate(0,-6.25px);
  transform: translate(0,-6.25px);
  position: absolute;
  top: 25px;
  left: 100px;
  -webkit-animation: animation-2 1s 0.75s infinite linear;
  animation: animation-2 1s 0.75s infinite linear;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.emotion-4 {
  width: 8.333333333333334px;
  height: 8.333333333333334px;
  background-color: #000000;
  margin: 2px;
  border-radius: 100%;
  -webkit-transform: translate(0,-6.25px);
  -ms-transform: translate(0,-6.25px);
  transform: translate(0,-6.25px);
  position: absolute;
  top: 25px;
  left: 100px;
  -webkit-animation: animation-2 1s 1s infinite linear;
  animation: animation-2 1s 1s infinite linear;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.emotion-5 {
  width: 8.333333333333334px;
  height: 8.333333333333334px;
  background-color: #000000;
  margin: 2px;
  border-radius: 100%;
  -webkit-transform: translate(0,-6.25px);
  -ms-transform: translate(0,-6.25px);
  transform: translate(0,-6.25px);
  position: absolute;
  top: 25px;
  left: 100px;
  -webkit-animation: animation-2 1s 1.25s infinite linear;
  animation: animation-2 1s 1.25s infinite linear;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

<Loader
  color="#000000"
  css=""
  loading={true}
  margin={2}
  size={25}
>
  <div
    className="emotion-6"
  >
    <div
      className="emotion-0"
    />
    <div
      className="emotion-1"
    />
    <div
      className="emotion-2"
    />
    <div
      className="emotion-3"
    />
    <div
      className="emotion-4"
    />
    <div
      className="emotion-5"
    />
  </div>
</Loader>
`;
